Katello/katello
# Fixes #36753 - Only allow coherent default CVE in global registration form

#### What are the changes introduced in this pull request?

Now that content views and lifecycle environments are assigned as a single unit (CVE / Content View Environment), it doesn't make sense to be able to select lifecycle environment separately. Also, the web UI was allowing you to select a non-library LCE and attempt to use it with the 'Default Organization View' content view, which won't work. So:

* Remove 'Lifecycle Environment' field from global registration form
* Add a Rails validator preventing content view environments from using a non-default lifecycle environment with the Default Organization View.

#### Considerations taken when implementing this change?

It's always been the case that you can't create a content view environment with Default Organization View and a non-library LCE. This just improves the error messaging and UI.

#### What are the testing steps for this pull request?

Before checking out the PR:

* Create an activation key assigning Library environment and Default Organization View
* Create at least one lifecycle environment
* Hosts > Register Host > Generate a registration command with the following properties:
  * Insecure
  * Activation keys: `<the one you created>`
  * Lifecycle Environment: `<the one you created>`
  * Force: true (more convenient if you're using an existing host)

Run the registration command on the host.  You'll get

```
The system has been registered with ID: 8b3a0960-9dbe-4428-9ac5-661e1ca892c3
The registered system name is: rhel8.fedora.example.com
ERROR: standard_error
PG::UniqueViolation: ERROR:  duplicate key value violates unique constraint index_cve_cp_id
DETAIL:  Key (cp_id)=(64c2cce3716a5bf67b0413fac95e5daa) already exists.
```

Now, check out the PR and use the _same_ generated registration command (this is where the "force" comes in handy :)

Now you'll get

```
The system has been registered with ID: 895e054c-9e47-42b2-a690-9c94b65bea1d
The registered system name is: rhel8.fedora.example.com
ERROR: Validation failed: Lifecycle environment 'dev2' cannot be used with content view 'Default Organization View'
```

Now go to the web UI - Hosts > Register Host

* Verify that the 'Lifecycle Environment' field is now gone
* verify that generating commands still works as expected
